python/cuml/comm/serialize.py

```diff
@@ -25,11 +27,37 @@
 serializable_classes = (cuml.common.CumlArray,)
 
 
+def serialize_mat_descriptor(m):
+    return cp.cusparse.MatDescriptor.create, ()
+
+
 try:
     from distributed.protocol import dask_deserialize, dask_serialize
     from distributed.protocol.cuda import cuda_deserialize, cuda_serialize
     from distributed.utils import log_errors
 
+    from distributed.protocol import register_generic
+
+    from cuml.naive_bayes.naive_bayes import MultinomialNB
+
+    try:
+
+        # cuda serializer requires Dask Distributed 2.12.0+
+        register_generic(MultinomialNB, 'cuda',
+                         cuda_serialize, cuda_deserialize)
+
+        register_generic(cuml.Base, 'cuda',
+                         cuda_serialize, cuda_deserialize)
+
+    except TypeError:
+        pass
+
+    # Registers 'dask' serialization by default
+    register_generic(MultinomialNB)
+    register_generic(cuml.Base)
```

Comment:
Was thinking of suggesting this, but didn't know if you had a reason not to do this. If you make this change, can drop 
`register_generic` calls on `MultinomialNB` if it is a subclass of `cuml.Base`. This goes both for this line and the one in the `try` section above.

Reply:
Unfortunatley, MultinomialNB doesn't extend `cuml.Base` so the separate registration is needed at this time. At one point I had a comment in here about this but between other refactors, merges, and moving stuff around in this PR, it was lost. 

I'll open up an issue and reference it here.

Reply:
Ah ok. Thanks for the context.

Do the other models contain attributes that we can already serialize or are there other things we are still missing?

Reply:
Naive Bayes is a little different than a lot of the other models because it 1) is 100% Python on the cuML side, and 2) accepts sparse arrays as input. Like some of the other models, however, it has not yet been refactored to use the new cuml array. This refactor will hopefully be conquered in 0.14, along with many others: https://github.com/rapidsai/cuml/issues/1834